scm-step/src/main/java/org/jenkinsci/plugins/workflow/steps/scm/GitStep.java

```diff
@@ -78,6 +119,57 @@ public DescriptorImpl() {
             GitSCM.class.hashCode();
         }
 
+        // copy/paste from GitSCM.DescriptorImpl
+
+        public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item project,
+                                                     @QueryParameter String url) {
+            if (project == null || !project.hasPermission(Item.CONFIGURE)) {
+                return new StandardListBoxModel();
+            }
+            return new StandardListBoxModel()
+                    .withEmptySelection()
+                    .withMatching(
+                            GitClient.CREDENTIALS_MATCHER,
+                            CredentialsProvider.lookupCredentials(StandardCredentials.class,
+                                    project,
+                                    ACL.SYSTEM,
+                                    GitURIRequirementsBuilder.fromUri(url).build())
+                    );
+        }
+
+        public FormValidation doCheckUrl(@AncestorInPath Item project,
+                                         @QueryParameter String credentialsId,
+                                         @QueryParameter String value) throws IOException, InterruptedException {
+
+            if (project == null || !project.hasPermission(Item.CONFIGURE)) {
+                return FormValidation.ok();
+            }
+
+            String url = Util.fixEmptyAndTrim(value);
+            if (url == null)
+                return FormValidation.error("Please enter Git repository.");
```

Comment:
nit: should be a sentence.  "Please enter the Git repository url." (or as this is already git in the context  - "Please enter repository url".